Sources/PrimerSDK/Classes/Extensions & Utilities/PrimerButton.swift

```diff
@@ -179,7 +180,9 @@ extension PrimerButton {
             setImage(UIImage(), for: state)
         }
         self.activityIndicatorButtonStates = buttonStates
-        isEnabled = false
+        DispatchQueue.main.async {
+            self.isEnabled = false
```

Comment:
I am not sure if this is needed, have you tested it? On the other hand, `activityIndicator.startAnimating()` on line 173 probably needs to be performed on the main queue.

Reply:
I've tested it yeah. Everything is actually performed on the main thread so that `DispatchQueue.main.async {` block shouldn't even be needed. Although in order to trigger the expected animation upon setting the `PrimerButton` disabled, that's the only solution I found to workaround the isssue.